advanced_functionality/autogluon-multimodal-containers/ag_model.py

```diff
@@ -0,0 +1,89 @@
+from sagemaker.estimator import Framework
+from sagemaker.predictor import Predictor
+from sagemaker.mxnet import MXNetModel
+from sagemaker.mxnet.model import MXNetPredictor
+from sagemaker import utils
+from sagemaker import image_uris
+from sagemaker.serializers import CSVSerializer
+from sagemaker.deserializers import StringDeserializer
+
+
+class AutoGluonTraining(Framework):
+    def __init__(
+        self,
+        entry_point,
+        region,
+        framework_version,
+        py_version,
+        instance_type,
+        source_dir=None,
+        hyperparameters=None,
+        **kwargs,
+    ):
+        self.framework_version = framework_version
+        self.py_version = py_version
+        self.image_uri = image_uris.retrieve(
+            "autogluon",
+            region=region,
+            version=framework_version,
+            py_version=py_version,
+            image_scope="training",
+            instance_type=instance_type,
+        )
+        super().__init__(
+            entry_point,
+            source_dir,
+            hyperparameters,
+            framework_version=framework_version,
+            instance_type=instance_type,
+            image_uri=self.image_uri,
+            **kwargs,
+        )
+
+    def _configure_distribution(self, distributions):
+        return
+
+    def create_model(
+        self,
+        model_server_workers=None,
+        role=None,
+        vpc_config_override=None,
+        entry_point=None,
+        source_dir=None,
+        dependencies=None,
+        image_name=None,
+        **kwargs,
+    ):
+        return None
+
+
+class AutoGluonMultimodalPredictor(Predictor):
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args, serializer=CSVSerializer(), deserializer=StringDeserializer(), **kwargs
+        )
+
+
+class AutoGluonInferenceModel(MXNetModel):
```

Comment:
[minor] Since MXNet is not used in AutoMM, is there a PyTorch counterpart we can leverage?

Reply:
There might be API changes between MXNet and Pytorch scripts. We also still us MXNet DLC containers.